mlir/test/Quantum/ChainedSelfInverseTest.mlir

```diff
@@ -37,3 +37,217 @@ func.func @test_chained_self_inverse(%arg: !quantum.bit) -> !quantum.bit {
     %1 = quantum.custom "Hadamard"() %0 : !quantum.bit
     return %1 : !quantum.bit
 }
+
```

Comment:
nice mlir tests! :100:

Reply:
The last test is exactly this :)

Reply:
Oops, I meant non-neighbouring self inverses are not canceled, e.g. `H-X-H`, all three gates should survive

Reply:
Added